cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java

```diff
@@ -409,7 +425,7 @@ private void scanDirectory(
         } else {
           String resourceName = packagePrefix + name;
           if (!resourceName.equals(JarFile.MANIFEST_NAME)) {
-            resources.add(ResourceInfo.of(resourceName, classloader));
+            resources.add(ResourceInfo.of(getFileBaseURL(directory, packagePrefix), resourceName, classloader));
```

Comment:
Yes. The packagePrefix is the package name of the given resource at the current level (it's recursive).